src/Build/Instance/TaskRegistry.cs

```diff
@@ -660,7 +682,22 @@ private void RegisterTask(string taskName, AssemblyLoadInfo assemblyLoadInfo, st
                 _taskRegistrations[taskIdentity] = registeredTaskEntries;
             }
 
-            registeredTaskEntries.Add(new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord));
+            RegisteredTaskRecord newRecord = new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord);
```

Comment:
The cool kids are doing it this way now (I don't actually care which you do):

```suggestion
            RegisteredTaskRecord newRecord = new(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord);
```